csrc/device_lower/pass/predicate.cpp

```diff
@@ -47,6 +47,14 @@ class ConditionalFromPredicateModifier : public kir::ExprMutator {
       // Replace expr predicate with bool conditional
       auto conditional = generateConditional(expr->predicate());
 
+      // When current elect sync predicates a UBLK TMA load, it also need to
```

Comment:
I generally don't think it's a good idea to expand this function. As indicated in the comment at line 60, the logic for the `Vectorize` is already not ideal either. This PR seems to make it much more complicated by the intertwined logic.

Ideally, this pass should be just converting `kir::Predicate` with `generateConditional`. Can we avoid doing structural changes like `reviseUblkPredicate` here and instead make KIR already properly structured before this pass?

Reply:
I think we can do that, but I feel it is moving the complexity from `generateConditionalFromPredicate` to `Unroll` pass